roxie/ccd/ccdcontext.cpp

```diff
@@ -3666,7 +3666,7 @@ class CRoxieServerContext : public CRoxieContextBase, implements IRoxieServerCon
     }
     IUserDescriptor *queryUserDescriptor()
     {
-        return NULL; // TBD - Richard, where do user credentials for a roxie query come from
+        return workUnit->queryUserDescriptor();
```

Comment:
I don't think you can assume there is a workunit - there will be in "ad-hoc" mode but not in predeployed query mode.